sc-tools/sc-builder/tests/units/test_gwf_translator.cpp

```diff
@@ -4,33 +4,209 @@
+  std::string const & gwfText = translator.GetXMLFileContent(filePath);
+  EXPECT_THROW(translator.TranslateGWFToSCs(gwfText, BASE_TEST_PATH), utils::ExceptionParseError);
+}
+
+TEST_F(GWFTranslatorTest, UnknownLinkContent)
+{
+  GWFTranslator translator(*m_ctx);
+
+  std::string const filePath = BASE_TEST_PATH "unknown_link_content.gwf";
+  std::string const & gwfText = translator.GetXMLFileContent(filePath);
+  EXPECT_THROW(translator.TranslateGWFToSCs(gwfText, BASE_TEST_PATH), utils::ExceptionParseError);
+}
+
+TEST_F(GWFTranslatorTest, ContourWithUnknownParent)
+{
+  GWFTranslator translator(*m_ctx);
+
+  std::string const filePath = BASE_TEST_PATH "contour_with_unknown_parent.gwf";
+  std::string const & gwfText = translator.GetXMLFileContent(filePath);
+  EXPECT_THROW(translator.TranslateGWFToSCs(gwfText, BASE_TEST_PATH), utils::ExceptionParseError);
+}
+
+TEST_F(GWFTranslatorTest, NodeWithUnknownId)
+{
+  GWFTranslator translator(*m_ctx);
+
+  std::string const filePath = BASE_TEST_PATH "node_with_unknown_id.gwf";
+  std::string const & gwfText = translator.GetXMLFileContent(filePath);
+  EXPECT_THROW(translator.TranslateGWFToSCs(gwfText, BASE_TEST_PATH), utils::ExceptionParseError);
+}
+
+TEST_F(GWFTranslatorTest, NodeWithUnknownType)
+{
+  GWFTranslator translator(*m_ctx);
+
+  std::string const filePath = BASE_TEST_PATH "element_with_unknown_type.gwf";
+  std::string const & gwfText = translator.GetXMLFileContent(filePath);
+  EXPECT_THROW(translator.TranslateGWFToSCs(gwfText, BASE_TEST_PATH), utils::ExceptionItemNotFound);
+}
+
+TEST_F(GWFTranslatorTest, Bus)
+{
+  EXPECT_TRUE(CheckGWFToSCSTranslation(m_ctx, "bus.gwf"));
+}
+
+TEST_F(GWFTranslatorTest, EmptyContour)
+{
+  EXPECT_TRUE(CheckGWFToSCSTranslation(m_ctx, "empty_contour.gwf"));
+}
+
+TEST_F(GWFTranslatorTest, LotOfContours)
+{
+  EXPECT_TRUE(CheckGWFToSCSTranslation(m_ctx, "lot_of_contours.gwf"));
+}
+
+TEST_F(GWFTranslatorTest, ContentTypes)
+{
+  EXPECT_TRUE(CheckGWFToSCSTranslation(m_ctx, "content_types.gwf"));
+
+  EXPECT_TRUE(CompareFiles(BASE_TEST_PATH "ostis.png", BASE_TEST_PATH "ostis_ref.png"));
+  std::filesystem::remove(BASE_TEST_PATH "ostis.png");
+}
+
+TEST_F(GWFTranslatorTest, SCsTree)
+{
+  GWFTranslator translator(*m_ctx);
+
+  std::string const & testGWFFilePath = BASE_TEST_PATH "empty_contour.gwf";
+  std::string const & correctGWFFilePath = BASE_TEST_PATH "lot_of_contours.gwf";
+  std::string const & correctSCsFilePath = correctGWFFilePath + ".scs";
+
+  std::string const & testGWFText = translator.GetXMLFileContent(testGWFFilePath);
+  std::string const & testSCsText = translator.TranslateGWFToSCs(testGWFText, BASE_TEST_PATH);
+
+  std::string const & correctSCsText = ReadFileToString(correctSCsFilePath);
+
+  SCsTree testSCsTree;
+  testSCsTree.Parse(testSCsText);
+  SCsTree correctSCsTree;
+  correctSCsTree.Parse(correctSCsText);
+  EXPECT_FALSE(testSCsTree.Compare(correctSCsTree)->empty());
+  EXPECT_FALSE(correctSCsTree.Compare(testSCsTree)->empty());
```

Comment:
I don't think that these two checks are helpful, they are just saying that trees are different but without specifying what those differences should be